### DISCUSSION TITLE
Does milvus has any plan to support Huawei Ascend chip?

### DISCUSSION THREAD

**Discussion by allengaoo** [milvus-io/milvus#35334] [Q&A and General discussion]
Does milvus support Huawei Ascend chip? If not ,  is there any plan for it?

**Comment by yhmo** [comment]
Milvus supports general x86 and arm64 CPU, we don't have plan to support specific hardware.
There are some discussions about milvus&ARM, you can read for more info:
https://github.com/milvus-io/milvus/discussions?discussions_q=is%3Aopen+arm

**Reply by SuJinpei** [reply]
Hi @yhmo, reaching out to see if there are any updates on Huawei Ascend support? I noticed some recent community interest and discussion in Issue #46224 regarding NPU optimizations. Has the project's roadmap evolved to include these community-led contributions, or is the focus still primarily on general x86/arm64? Thanks!

**Reply by yhmo** [reply]
> Hi @yhmo, reaching out to see if there are any updates on Huawei Ascend support? I noticed some recent community interest and discussion in Issue #46224 regarding NPU optimizations. Has the project's roadmap evolved to include these community-led contributions, or is the focus still primarily on general x86/arm64? Thanks!

Actually, we don't have any plan to support a specific NPU. It would require a fundamentally different compute backend (e.g.,  a vendor-specific SDK). This would be a significant engineering effort that we don't have engineers with the expertise. Anyway, community contribution is always welcome.
